### PR TITLE
upgrade realm-js to 2.16.1

### PR DESCRIPTION
**Pull request by rasom**
includes @mandrigin's PR https://github.com/realm/realm-js/pull/2020, so no need to patch realm-js during builds
 
status: ready

### CODE REVIEW

**Review by janherich — APPROVED** [review]

**Review by mandrigin — APPROVED** [review]
Lgtm!

**Review by yenda — APPROVED** [review]

File: mobile_files/package.json [hunk 1]
```diff
@@ -61,8 +62,7 @@
     "react-native-udp": "https://github.com/status-im/react-native-udp.git#2.3.1-1",
     "react-native-webview-bridge": "git+https://github.com/status-im/react-native-webview-bridge.git#0.33.16-1",
     "react-navigation": "^2.12.1",
-    "react-native-safe-area-view": "0.9.0",
-    "realm": "2.15.3",
+    "realm": "^2.16.1",
```
Comment (yenda) [inline mobile_files/package.json]: No ^ we need fixed releases for reproducible builds.
Also they release a minor version in the past without android support

**Review by yenda — CHANGES_REQUESTED** [review]
use fixed version number

File: mobile_files/package.json (hunk 1 above)
Comment (rasom) [reply mobile_files/package.json]: fixed

**Review by yenda — APPROVED** [review]

**Pull request merged**